src/cirq_iqm/devices/iqm_device.py

```diff
@@ -51,48 +53,41 @@ class IQMDevice(devices.Device):
 
     Adds extra functionality on top of the basic :class:`cirq.Device` class for decomposing gates,
     optimizing circuits and mapping qubits.
-    """
-
-    QUBIT_COUNT: Optional[int] = None
-    """number of qubits on the device"""
-
-    QUBIT_NAME_PREFIX: str = 'QB'
-    """prefix for qubit names, to be followed by their numerical index"""
-
-    CONNECTIVITY: tuple[set[int], ...] = ()
-    """qubit connectivity graph of the device"""
 
-    NATIVE_GATES: tuple[type[cirq.Gate], ...] = (ops.PhasedXPowGate, ops.XPowGate, ops.YPowGate, ops.MeasurementGate)
-    """native gate set of the device (gate families)"""
+    Args:
+        metadata: device metadata which contains the qubits, their connectivity, and the native gateset
+    """
 
-    NATIVE_GATE_INSTANCES: tuple[cirq.Gate, ...] = (ops.CZPowGate(),)
-    """native gate set of the device (individual gates)"""
+    def __init__(self, metadata: IQMDeviceMetadata):
+        self._metadata = metadata
+        self.qubits = tuple(sorted(self._metadata.qubit_set))
```

Comment:
The sorting here will probably sort `QB10` before `QB2`?

Reply:
Could the metadata store a tuple of qubits instead of a set, and have the property convert it into a frozenset on demand? The tuple already has the correct order.

Reply:
>The sorting here will probably sort QB10 before QB2

No, `NamedQubits` with numbers are sorted correctly (https://quantumai.google/reference/python/cirq/NamedQubit: *NamedQubit has a lexicographic order. However, numbers within the name are handled correctly...*). I tested also by printing the `qubits` of Apollo. 

Since this is the case, I would not store them as tuple unless there is some other reason